javascript/view.js

```diff
@@ -0,0 +1,145 @@
+const displayProductsView = {
+    init(DISPLAY_PRODUCTS_CONTENT){
+        this.productsToBeSold = document.getElementById('productsToBeSold');
+        this.DISPLAY_PRODUCTS_CONTENT = DISPLAY_PRODUCTS_CONTENT;
+        //console.log(this.DISPLAY_PRODUCTS_CONTENT[0].categories);
+        this.render();
+    },
+
+    render(){
+        for(this.i=0; this.i < this.DISPLAY_PRODUCTS_CONTENT.length; this.i++)
+        {
+            this.assignElement();
+
+            this.assignClassName();
+
+            this.assignInnerContent();
+
+            this.appendingToElements();
+        }
+    },
+
+    assignElement(){
+        this.offer = document.createElement('div');
```

Comment:
Rather than dividing the function like this, create separate functions for creating different kind of elements. This way all the logic related to creating that particular element will get encapsulated in a single function.